### ISSUE TITLE
AWS S3 SDK doesn't support file renaming. 

### ISSUE THREAD

**Issue by nazeer1100126** [aws/aws-sdk-java#2206]
There is no method available to rename a particular file in AWS S3 SDK.
The suggested approach copying the file content to new file is time consuming.

**Comment by nazeer1100126** [comment]
com.amazonaws.services.s3.transfer.TransferManager

**Comment by debora-ito** [comment]
No, there's no rename operation available in S3. This is a feature request for the S3 API team, so I will reach out to them and report your request. 

I would also suggest raising this in the AWS Developer Forum 
https://forums.aws.amazon.com/forum.jspa?forumID=24 or via customer support, to engage the service team directly.

**Comment by nazeer1100126** [comment]
Thank you @debora-ito for taking initiative to reach out API team.

**Comment by debora-ito** [comment]
For visibility to other SDKs, I'm moving this to the `aws-sdk` repository.
